### DISCUSSION TITLE
VS Code' Explorer menu item is not added

### DISCUSSION THREAD

**Discussion by GauthierPLM** [microsoft/winget-pkgs#2189] [General]
Installing VS Code using Winget does not add the menu item "open file in VS Code" to Explorer while it is added by default using the standalone installer.

As it is a very convenient feature, and other software like 7-Zip are added to the menu, it would be great to have this menu item added.

**Comment by Xeevis** [comment]
By default only PATH is checked so silent installer does in fact work correctly.

![image](https://user-images.githubusercontent.com/5835044/86159107-18e5c580-bb0a-11ea-87a8-5d73c6760288.png)

You can override this behavior by passing in additional parameters

> winget install Microsoft.VisualStudioCode -e --override "/verysilent /suppressmsgboxes /tasks=!runCode,desktopicon,quicklaunchicon,addcontextmenufiles,addcontextmenufolders,associatewithfiles,addtopath"

Prepending tasks with `!` disables them.

**Comment by GauthierPLM** [comment]
This is exactly what I am looking for; it works like a charm.

However, I think without any explanation, most users would miss this option. I don't even know how to get this list of tasks.

**Comment by Xeevis** [comment]
Tradeoff for silent install, not much that can be done unless VS Code team decides these default values should be changed. Tasks can be found in VS Code setup manifest. https://github.com/microsoft/vscode/blob/aa98d29af94188825cbe2cf32864f1f33dc90b6e/build/win32/code.iss#L75-L82
